### PR TITLE
Moves GHA docker invocation from action to Makefile

### PR DESCRIPTION
**Pull request by tcsc**
Consolidates more of the build logic into the build.assets `Makefile`, transplanted from the workflow file in `teleport.e`

See comment https://github.com/gravitational/teleport.e/pull/673#discussion_r1059530522

### CODE REVIEW

**Review by camscale — APPROVED** [review]
LGTM. Some comments you can take or leave. Also, I congratulate you on two different misspellings of the word "supplied". :rofl:

File: build.assets/Makefile [hunk 1]
```diff
@@ -434,19 +434,50 @@ release-arm64: buildbox-arm
...
+		-v "$(WORKSPACE):/go/teleport" \
+		-w /go/teleport                \
+		-u 1000:1000                   \
+		$(BUILDBOX)					   \
+		make full build-archive -e ARCH=$(ARCH) ADDFLAGS="$(ADDFLAGS)" OS=$(OS) RUNTIME=$(GOLANG_VERSION) FIDO2=$(FIDO2) PIV=$(PIV) REPRODUCIBLE=no
 
 .PHONY: release-enterprise
 release-enterprise:
-	make -C .. release-unix-only -e ARCH=$(ARCH) ADDFLAGS="$(ADDFLAGS)" OS=$(OS) RUNTIME=$(GOLANG_VERSION) FIDO2=$(FIDO2) PIV=$(PIV) REPRODUCIBLE=no
+	docker run                         \
```
[the remark this answers is not in the document]
Comment (tcsc) [reply build.assets/Makefile:468]: Yeah, my plan is to condense this into something smarter once I have the actual build up and running. My focus right now is on the interface between GHA and this makefile. (if that makes sense?)

File: build.assets/Makefile [hunk 2]
```diff
@@ -434,19 +434,50 @@ release-arm64: buildbox-arm
...
+#
+# Separating the constrtuction of build environment from the actual build itself 
+# allows us to use the appropriate GHA actions to pre-build the buildbox 
+# image, so that the image construction works nicely with GHA.
 #
 
 .PHONY: release-oss
 release-oss:
-	make -C .. full build-archive -e ARCH=$(ARCH) ADDFLAGS="$(ADDFLAGS)" OS=$(OS) RUNTIME=$(GOLANG_VERSION) FIDO2=$(FIDO2) PIV=$(PIV) REPRODUCIBLE=no
+	docker run                         \
```
[the remark this answers is not in the document]
Comment (tcsc) [reply build.assets/Makefile:452]: I get that, but (to me, at least) it also leaves the bits you care about easier to read. I can stop doing it though  - not a hill worth dying on for me.

File: build.assets/Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (tcsc) [reply build.assets/Makefile:468]: Actually, I like the define. Let's try that.

**Review by strideynet — APPROVED** [review]

**Pull request merged**